### PR TITLE
Replace short name that is between [' or"]

### PR DESCRIPTION
**Pull request by mariarivera30**
- Replace all instances of old_name in the task presenter HTML with new_name when cloning a project.

### CODE REVIEW

File: pybossa/view/projects.py [hunk 1]
```diff
@@ -449,13 +449,15 @@ def clone_project(project, form):
...
     proj_dict['name'] = form['name']
-    replacement = 'pybossa.run("%s")' % proj_dict['short_name']
-    pybossa_re = "(pybossa\.run\(\s*)(['\"]\S+['\"]\))"
-    proj_dict['info']['task_presenter'] = re.sub(pybossa_re,
-                                                 replacement,
-                                                 proj_dict['info'].get('task_presenter', ''))
+
+    task_presenter = proj_dict['info'].get('task_presenter', '')
+    for quoted_part in re.findall(r"[\'\"](.*?)[\"\']", task_presenter):
+        task_presenter = task_presenter.replace(quoted_part, quoted_part.replace(proj_dict['short_name'], form['short_name']))
```
Comment (fiorda) [inline pybossa/view/projects.py]: You might be able to use capture groups and back references here - https://docs.python.org/2/library/re.html#re.sub

Comment (mariarivera30) [reply pybossa/view/projects.py]: As discussed will just match  when the short name appears in "(name)" or '(name) ' or /(name)/

File: pybossa/view/projects.py [hunk 2]
```diff
@@ -449,13 +449,14 @@ def clone_project(project, form):
...
-    proj_dict['short_name'] = form['short_name']
     proj_dict['name'] = form['name']
-    replacement = 'pybossa.run("%s")' % proj_dict['short_name']
-    pybossa_re = "(pybossa\.run\(\s*)(['\"]\S+['\"]\))"
-    proj_dict['info']['task_presenter'] = re.sub(pybossa_re,
-                                                 replacement,
-                                                 proj_dict['info'].get('task_presenter', ''))
+
+    task_presenter = proj_dict['info'].get('task_presenter', '')
+    regex = r"([\"\'\/])({})([\"\'\/])".format(proj_dict['short_name'])
```
Comment (primaryobjects) [inline pybossa/view/projects.py]: This looks good and works well. Just to make the code a little more clear, one item to consider is naming the variables, and perhaps a comment to explain the regex? :)

```python
old_name = proj_dict['short_name']
new_name = form['short_name']

# Replace all instances of old_name in the task presenter HTML with new_name.
regex = r"([\"\'\/])({})([\"\'\/])".format(old_name)
proj_dict['info']['task_presenter'] = re.sub(regex, r"\1{}\3".format(new_name), task_presenter)

Comment (kbecker42) [inline pybossa/view/projects.py]: This looks good and works well. Just to make the code a little more clear, one item to consider is naming the variables, and perhaps a comment to explain the regex? :)

```python
old_name = proj_dict['short_name']
new_name = form['short_name']

# Replace all instances of old_name in the task presenter HTML with new_name.
regex = r"([\"\'\/])({})([\"\'\/])".format(old_name)
proj_dict['info']['task_presenter'] = re.sub(regex, r"\1{}\3".format(new_name), task_presenter)

File: pybossa/view/projects.py [hunk 3]
```diff
@@ -449,13 +449,14 @@ def clone_project(project, form):
...
+    proj_dict['info']['task_presenter'] = re.sub(regex, r"\1{}\3".format(form['short_name']), task_presenter)
+
+    proj_dict['short_name'] = form['short_name']
```
Comment (kbecker42) [inline pybossa/view/projects.py]: If using variables per above, this could probably be changed as well.

```python
proj_dict['short_name'] = new_name
```

**Pull request merged**